lib/kube/proxy/forwarder.go

```diff
@@ -2770,3 +2776,25 @@ func errorToKubeStatusReason(err error, code int) metav1.StatusReason {
 		return metav1.StatusReasonUnknown
 	}
 }
+
+// formatForwardResponseError formats the error response from the connection
+// monitor to a Kubernetes API error response.
+type formatForwardResponseError func(status *apierrors.StatusError) error
+
+func (f formatForwardResponseError) WriteString(s string) (n int, err error) {
+	if f == nil {
+		return len(s), nil
+	}
+	_ = f(
+		&apierrors.StatusError{
+			ErrStatus: metav1.Status{
+				Status:  metav1.StatusFailure,
+				Code:    http.StatusInternalServerError,
+				Reason:  metav1.StatusReasonInternalError,
+				Message: s,
+			},
+		},
+	)
+
+	return len(s), nil
```

Comment:
Should f return (int, error), so we get a more accurate count? Does it matter?

If no then I think it's worth having a comment here.

Reply:
adjusted